📝 [RUMF-1469] document new proxy configuration (#16727)
DataDog/documentation: content/en/real_user_monitoring/guide/proxy-rum-data.md

## Proxy setup

> When your proxy receives data from the RUM Browser SDK, it must be forwarded to Datadog. The RUM Browser SDK adds the `ddforward` query parameter to all POST requests to your proxy. This query parameter contains the URL path and parameters where all data must be forwarded to.

Reviewer: ```suggestion
When your proxy receives data from the RUM Browser SDK, it must be forwarded to Datadog. The RUM Browser SDK adds the `ddforward` query parameter to all POST requests to your proxy. This query parameter contains the URL path and parameters that all data must be forwarded to.
```

To successfully proxy request to Datadog:

1. Build the final Datadog intake URL by using:
    - the Datadog intake origin corresponding to your site. See table below.
    - the path and parameters provided by `ddforward`.
2. Add a `X-Forwarded-For` header containing the request client IP address for accurate geoIP.
3. Forward the request to the Datadog intake URL using the POST method.

> 4. The request body must remain unchanged.

Reviewer: ```suggestion
4. Leave the request body unchanged.
```

> The Site Parameter is an SDK [initialization parameter][1]. Datadog intake origins for each site are listed below:

Reviewer: ```suggestion
The site parameter is an SDK [initialization parameter][1]. Datadog intake origins for each site are listed below:
```

| Site    | Site Parameter            | Datadog intake origin                      |
| ------- | ------------------------- | ------------------------------------------ |
| US1     | `datadoghq.com` (default) | `https://browser-intake-datadoghq.com`     |
| US3     | `us3.datadoghq.com`       | `https://browser-intake-us3-datadoghq.com` |
| US5     | `us5.datadoghq.com`       | `https://browser-intake-us5-datadoghq.com` |
| EU1     | `datadoghq.eu`            | `https://browser-intake-datadoghq.eu`      |
| US1-FED | `ddog-gov.com`            | `https://browser-intake-ddog-gov.com`      |

> The Datadog intake origin corresponding to your Site Parameter should be defined in your proxy implementation.

Reviewer: ```suggestion
The Datadog intake origin corresponding to your site parameter should be defined in your proxy implementation.
```

## Legacy proxy setups

Before Browser SDK v4.TODO, the `proxyUrl` initialization parameter was used and the Datadog intake origin was included in the <code>ddforward</code> attribute. The proxy implementation was in charge of validating this host and failure to do so resulted in various vulnerabilities.

The Datadog intake origin now needs to be defined in your proxy implementation to ensure security.

> <strong>If you are still using a proxy with an older version of the Browser SDK, please consider upgrading to a newer version of the Browser SDK.</strong>

Reviewer: ```suggestion
<strong>If you are still using a proxy with an older version of the Browser SDK, we recommend upgrading to a newer version of the Browser SDK.</strong>
```
